source/declarations.tex

```diff
@@ -2562,37 +2551,33 @@
 \end{example}
 
 \pnum
-In a declaration
-\opt{\grammarterm{attribute-specifier-seq}}
-\tcode{T}
-\tcode{D}
-where
-\tcode{D}
-is an unadorned identifier the type of this identifier is
-``\tcode{T}''.
+In a declaration \tcode{T} \tcode{D}
+where \tcode{D} has the form
+\begin{ncsimplebnf}
+declarator-id \opt{attribute-specifier-seq}
+\end{ncsimplebnf}
+the type of the \grammarterm{declarator-id}
+in \tcode{D} is ``\tcode{T}''.
+The optional \grammarterm{attribute-specifier-seq}
+appertains to the entity declared by \tcode{D}.
 
 \pnum
 In a declaration
-\tcode{T}
-\tcode{D}
-where
-\tcode{D}
-has the form
+\tcode{T} \tcode{D} where
+\tcode{D} has the form
 \begin{ncsimplebnf}
 \terminal{(} \terminal{D1} \terminal{)}
 \end{ncsimplebnf}
-the type of the contained
-\grammarterm{declarator-id}
-is the same as that of the contained
-\grammarterm{declarator-id}
-in the declaration
-\begin{codeblock}
-T D1
-\end{codeblock}
+the type of the \grammarterm{declarator-id}
+in \tcode{D} is the same as that of the
+contained \grammarterm{declarator-id}
+in the declaration \tcode{T} \tcode{D1}.
 \indextext{declaration!parentheses in}%
+\begin{note}
 Parentheses do not alter the type of the embedded
 \grammarterm{declarator-id},
 but they can alter the binding of complex declarators.
+\end{note}
```

Comment:
> A *declarator* of the form `(` *D* `)` determines the type of its declared entity to be the same as the type determined by the *declarator* *D*. [Note: ...]

Reply:
The problem is that a declarator (alone) does not determine any type at all. Example: `(*p)` is a declarator, but doesn't actually say what the type is: "pointer to whatever" is not a type. The rephrasing here departs from the careful constructions involved derived-declarator-type-list.